modules/imgproc/test/test_distancetransform.cpp

```diff
@@ -283,4 +283,28 @@ void CV_DisTransTest::prepare_to_validation( int /*test_case_idx*/ )
 
 TEST(Imgproc_DistanceTransform, accuracy) { CV_DisTransTest test; test.safe_run(); }
 
+BIGDATA_TEST(Imgproc_DistanceTransform, large_image_12218)
+{
+    int i,j;
+    Size size(20000, 8000);
+    Mat src(size, CV_8UC1, Scalar::all(128));
+    Mat ans(size, CV_32FC1, Scalar::all(8192));
```

Comment:
> 8192

Magic number from algorithm implementation?

Reply:
@alalek Thanks for your comments. This test is based on "testDistanceTransformWithLabels ()". I created this test with reference to imgproc / misc / java / test / ImgprocTest.java.

Reply:
@alalek This test case is not suitable,  so I will fix it later.